File: src/forms/form_buttons.py
"""
Extraction utilities for button controls.
"""

def extract_buttons(container):
    """Extract button elements and input elements with type button/submit/reset"""
    buttons = container.find_all('button')
    button_inputs = container.find_all('input', type=lambda t: t in ['button', 'submit', 'reset'])
    
    result = []
    
    for button in buttons:
        result.append({
            'id': button.get('id', ''),
            'name': button.get('name', ''),
            'type': button.get('type', 'button'),
            'text': button.text.strip(),
            'disabled': button.has_attr('disabled')
        })
    
    for button in button_inputs:
        result.append({
            'id': button.get('id', ''),
            'name': button.get('name', ''),
            'type': button.get('type', 'button'),
            'value': button.get('value', ''),
            'disabled': button.has_attr('disabled')
        })
        
    return result

File: browser-extension/forms/form_buttons.js
/**
 * Extraction utilities for button controls.
 */

/**
 * Extract button elements and input elements with type button/submit/reset
 * @param {Object} $ - Cheerio instance
 * @param {Object} container - Container element
 * @returns {Array} Array of button objects
 */
function extractButtons($, container) {
  const buttons = [];
  
  // Get standard buttons
  $(container).find('button').each((i, button) => {
    buttons.push({
      id: $(button).attr('id') || '',
      name: $(button).attr('name') || '',
      type: $(button).attr('type') || 'button',
      text: $(button).text().trim(),
      disabled: $(button).attr('disabled') !== undefined
    });
  });
  
  // Get input buttons
  $(container).find('input[type="button"], input[type="submit"], input[type="reset"]').each((i, button) => {
    buttons.push({
      id: $(button).attr('id') || '',
      name: $(button).attr('name') || '',
      type: $(button).attr('type') || 'button',
      value: $(button).attr('value') || '',
      disabled: $(button).attr('disabled') !== undefined
    });
  });
  
  return buttons;
}

module.exports = { extractButtons };
